### PR TITLE
Add first implementation of a suite tree view

### PR DESCRIPTION
**Pull request by kinow**
Using [vue-ads-table-tree](https://github.com/arnedesmedt/vue-ads-table-tree/), and displaying data from GraphQL query (need to use current WIP branch from David). Also works with offline mode for dev/testing.

### CODE REVIEW

File: src/transformers/vueadstabletree.transformer.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+  transform (graphqlFamilyProxies = []) {
+    return _.map(graphqlFamilyProxies, (familyProxy) => {
+      return this._recursivelyTransformFamily(familyProxy)
+    })
+  }
+
+  _recursivelyTransformFamily (familyProxy) {
+    const childrenFamilies = familyProxy.childFamilies.length > 0 ? _.map(familyProxy.childFamilies, this._recursivelyTransformFamily) : []
+    const childrenTasks = familyProxy.childTasks !== undefined && familyProxy.childTasks.length > 0 ? _.map(familyProxy.childTasks, this._transformTask) : []
```
Comment (matthewrmshin) [inline src/transformers/vueadstabletree.transformer.js]: Fine to call map with zero-length array?

File: src/views/Suite.vue [hunk 2]
```diff
@@ -71,50 +51,77 @@
...
         }
-      ]
+      ],
+      classes: {
+        table: "v-table",
+        '0/all': {'column text-xs-left': true}
+      },
+      filter: '',
+      start: 0,
+      end: 100,
```
Comment (matthewrmshin) [inline src/views/Suite.vue]: Comment on pagination?

File: src/views/Suite.vue [hunk 3]
```diff
@@ -71,50 +51,77 @@
...
+      filterChanged (filter) {
+        this.filter = filter;
+      },
+      fetchSuite() {
+        const suiteId = this.$route.params.name
+        suiteService.fetchSuiteTree(suiteId)
+        this.polling = setInterval(() => {
+          suiteService.currentTaskIndex += 1
+          suiteService.fetchSuiteTree(suiteId)
+        }, 3000)
```
Comment (matthewrmshin) [inline src/views/Suite.vue]: Comment on `3000ms` will be replaced by WebSocket.

File: src/views/Suites.vue [hunk 4]
```diff
@@ -116,6 +117,12 @@ export default {
   },
   beforeCreate() {
     suiteService.getSuites()
+    this.polling = setInterval(() => {
+      suiteService.getSuites()
+    }, 5000)
```
Comment (matthewrmshin) [inline src/views/Suites.vue]: Comment on another time interval that will be replaced by WebSocket.

File: src/transformers/vueadstabletree.transformer.js (hunk 1 above)
Comment (kinow) [reply src/transformers/vueadstabletree.transformer.js]: Yup, it just returns an empty array (should be covered in the unit tests)

File: src/views/Suite.vue (hunk 2 above)
Comment (kinow) [reply src/views/Suite.vue]: +1

File: src/views/Suite.vue (hunk 3 above)
Comment (kinow) [reply src/views/Suite.vue]: +1

File: src/views/Suites.vue (hunk 4 above)
Comment (kinow) [reply src/views/Suites.vue]: +1

File: src/transformers/vueadstabletree.transformer.js [hunk 5]
```diff
@@ -0,0 +1,56 @@
+import _ from 'lodash'
+import Family from '@/model/Family.model'
+import Task from '@/model/Task.model'
+
+export default class VueAdsTableTreeTransformer {
+
+  constructor () {
+
+  }
```
[the remark this answers is not in the document]
Comment (kinow) [reply src/transformers/vueadstabletree.transformer.js]: Done!

**Comment by kinow** [comment]
Feedback addressed, thanks @matthewrmshin !

**Comment by kinow** [comment]
Thanks @matthewrmshin !

**Comment by sadielbartholomew** [comment]
Great work @kinow (:deciduous_tree:-mendous)!

**Pull request merged**